File: sha256.py
import hashlib

prefix = 'zero2hero'
nonce = 0
twoZero = False

while True:
    input_str = prefix + str(nonce)
    hash_str = hashlib.sha256(input_str.encode()).hexdigest()
 
    if hash_str.startswith('00000'):
        print(f"Found nonce: {nonce}")
        print(f"Hash value: {hash_str}")
        break   
    elif hash_str.startswith('00') and not twoZero:
        print(f"Found nonce: {nonce}")
        print(f"Hash value: {hash_str}")
        twoZero = True

    nonce += 1

File: rust/sha256.rs
use crypto::digest::Digest;
use crypto::sha2::Sha256;

fn find_hash() {
    let mut nonce: u64 = 0;
    loop {
        let mut hasher = Sha256::new();
        let message = format!("zero2hero+{}", nonce);
        hasher.input_str(&message);
        let hash_result = hasher.result_str();
        if hash_result.starts_with("00") {
            println!("Found hash: {}", hash_result);
            return;
        }
        nonce += 1;
    }
}